hoomd/pytest/test_simulation.py

```diff
@@ -115,14 +115,21 @@ def test_run(simulation_factory, lattice_snapshot_factory):
     assert sim.operations._scheduled
 
 
-def test_tps(simulation_factory, lattice_snapshot_factory):
+def test_tps(simulation_factory, two_particle_snapshot_factory):
     sim = simulation_factory()
     assert sim.tps is None
 
-    sim = simulation_factory(lattice_snapshot_factory())
-    sim.run(100)
+    sim = simulation_factory(two_particle_snapshot_factory())
+    assert sim.tps == 0
+
+    sim.run(10)
+    tps = sim.tps
+    assert tps > 0
+    assert tps == sim.tps
 
-    assert sim.tps > 0
+    # Test that tps updates
+    sim.run(10)
+    assert sim.tps != tps
```

Comment:
Doesn't this have a tiny chance of failing?
Can we do something like increase the r_cut on the 2nd run to make sure it runs slower?

Reply:
Yes, there is a small chance this could fail. There are no particle interactions and no r_cut to change in this test. To ensure different timings, add a custom action that sleep different amounts of time in the two tests.

Reply:
I now run 10 steps and only require that at least 2 TPS values exist. I feel this is so unlikely not to occur with the given clock to the degree of floating point accuracy that this should be robust. If you disagree though, I will add a custom action (or if tests prove me wrong :sweat_smile: )